server/src/main/java/org/apache/iotdb/db/mpp/sql/planner/plan/node/write/InsertRowNode.java

```diff
@@ -172,17 +173,16 @@ public boolean validateSchema(SchemaTree schemaTree) {
         return false;
       }
     } else {
-      // todo partial insert
       if (deviceSchemaInfo.isAligned() != isAligned) {
         return false;
       }
 
       for (int i = 0; i < measurementSchemas.size(); i++) {
         if (dataTypes[i] != measurementSchemas.get(i).getType()) {
-          if (IoTDBDescriptor.getInstance().getConfig().isEnablePartialInsert()) {
+          if (!IoTDBDescriptor.getInstance().getConfig().isEnablePartialInsert()) {
```

Comment:
add a test case

Reply:
UT or IT?

Reply:
We don't have IT for new cluster yet. For now, UT is enough.